### ISSUE TITLE
HTML5 and CSS: Prioritize One Style Over Another

### ISSUE THREAD

**Issue by BlauFeuer** [FreeCodeCamp/FreeCodeCamp#10316]
Challenge [Prioritize One Style Over Another](https://www.freecodecamp.com/challenges/prioritize-one-style-over-another#?solution=%0A%3Cstyle%3E%0A%20%20body%20%7B%0A%20%20%20%20background-color%3A%20black%3B%0A%20%20%20%20font-family%3A%20Monospace%3B%0A%20%20%20%20color%3A%20green%3B%0A%20%20%7D%0A%20%20.pink-text%20%7Bcolor%3Apink%7D%0A%3C%2Fstyle%3E%0A%3Ch1%20class%3D%27pink-text%27%3EHello%20World!%3C%2Fh1%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:51.0) Gecko/20100101 Firefox/51.0</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```html

<style>
  body {
    background-color: black;
    font-family: Monospace;
    color: green;
  }
  .pink-text {color:pink}
</style>
<h1 class='pink-text'>Hello World!</h1>

```

Test requires semicolon separator, but since there is no additional class style element, it is not required.
Thus one can correctly style the output, but fail the test.

**Comment by dhcodes** [comment]
If I understand correctly, you are saying that `color: pink` doesn't require a semicolon at the end because there are no styles following it, but the semicolon is part of CSS syntax. Regardless of if it displays in the sample to the right of the code editor, the test necessitates it because it is the proper way to write the CSS.

See the examples near the bottom of this page: https://developer.mozilla.org/en-US/docs/Web/CSS/color

**Comment by BlauFeuer** [comment]
This might be updated, but it was not part of the standard when I learned it:
https://www.w3.org/TR/CSS2/syndata.html#declaration

The browser also displays it correctly; the test should only test for the presence of the `color: pink`.
Conformance to standards is not the same as your opinion on the proper way, but if you want to argue best practices, then rewrite the test constraint requirement to use what Free Code Camp believes is best practices.

**Comment by dhcodes** [comment]
Fair point and I learned something new.

**Comment by erictleung** [comment]
@BlauFeuer thanks for the issue and your points on why the test should be changed.

The standard does imply that you don't need semicolons after the last declaration, which is the case in this single declaration CSS rule. However, for consistency, we want to keep the semicolons there.

Someone has previously brought up this issue #8274. The [decision](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/8274#issuecomment-214069346) from @BerkeleyTrue on the matter is

> Yes, semicolons should always be included for consistency.

[Here](http://stackoverflow.com/questions/11939595/leaving-out-the-last-semicolon-of-a-css-block) are some more arguments on leaving out or keeping the last semicolon.

Closing as we have resolved this issue before. I hope you understand our decision. Happy coding!
